skimage/segmentation/_watershed.py

```diff
@@ -91,8 +93,11 @@ def _validate_inputs(image, markers, mask, connectivity):
             mask.astype(np.int8))
 
 
-def watershed(image, markers=None, connectivity=1, offset=None, mask=None,
-              compactness=0, watershed_line=False):
+def watershed(image: ImagendArray, markers: Optional[LabelsArray] = None,
+              connectivity: Union[int,_SupportsArray] = 1,
+              offset: Optional[np.ndarray] = None,
+              mask: Optional[MaskArray] = None,
+              compactness: float = 0, watershed_line: bool = False):
```

Comment:
I think if we support inline typing, we should prefer black's formatting style, which in this case means:

```suggestion
def watershed(
    image: ImagendArray,
    markers: Optional[LabelsArray] = None,
    connectivity: Union[int,_SupportsArray] = 1,
    offset: Optional[np.ndarray] = None,
    mask: Optional[MaskArray] = None,
    compactness: float = 0,
    watershed_line: bool = False,
) -> LabelsArray:
```

The ideas are:

- one argument per line
- always a trailing comma
- one indent level
- trailing parens on its own line

This scheme means minimal diffs when doing things like reordering arguments etc. I don't actually love it for untyped functions, where the argument signatures can fit much more compactly, but once you add types, it's a lot clearer to have one argument per line.

Also, `markers` can also be an int these days. =)